mslib/mscolab/mscolab.py

```diff
@@ -92,6 +94,72 @@ def handle_db_seed():
     seed_data()
     print("Database seeded successfully!")
 
+def handle_mscolab_certificate_init():
+    print('generating CRTs for the mscolab server......')
+    cmd = f"openssl req -newkey rsa:4096 -keyout {mscolab_settings.MSCOLAB_SSO_DIR}/key_mscolab.key -nodes -x509 -days 365 -batch -subj '/CN=localhost' -out {mscolab_settings.MSCOLAB_SSO_DIR}/crt_mscolab.crt"
+    os.system(cmd)
+    print('CRTs generated successfully for the mscolab server......')
+
+def handle_local_idp_certificate_init():
+    print('generating CRTs for the local identity provider......')
+    cmd = f"openssl req -newkey rsa:4096 -keyout {mscolab_settings.MSCOLAB_SSO_DIR}/key_local_idp.key -nodes -x509 -days 365 -batch -subj '/CN=localhost' -out {mscolab_settings.MSCOLAB_SSO_DIR}/crt_local_idp.crt"
+    os.system(cmd)
+    print('crts generated successfully for the mscolab local identity provider......')
+
+def handle_mscolab_metadata_init():
+    '''
+        This will generate necessary metada data file for sso in mscolab through localhost idp
+
+        Before running this function:
+        - Ensure that IDP_ENABLED is set to True.
+        - Generate the necessary keys and certificates and configure them in the .yaml 
+        file for the local IDP.
+    '''
+    print('generating metadata file for the mscolab server')
+
+    cmd ="python mslib/mscolab/mscolab.py start"
+    subprocess.Popen(cmd, shell=True)
```

Comment:
```suggestion
    subprocess.Popen(["python", "mslib/mscolab/mscolab.py", "start"])
```

While I don't see an issue with `shell=True` here, apart from unnecessary overhead, I also don't think it is necessary. In general `shell=True` can quickly become a security issue due to shell injection attacks (<https://docs.python.org/3/library/subprocess.html#security-considerations>), so I would like to avoid it as much as possible.